### DISCUSSION TITLE
By pass "Can't build edge function X: Exceeds maximum edge function size: 4.45 MB / 4 MB"

### DISCUSSION THREAD

**Discussion by SamyPesse** [cloudflare/next-on-pages#830] [General]
We have deployments failing since we updated a library (https://github.com/GitbookIO/gitbook/pull/2338). This library is lazy loaded so in theory it should not impact the worker initial load size.

```
Error: Can't build edge function /(space)/(content)/[[...pathname]]/page: Exceeds maximum edge function size: 4.45 MB / 4 MB
```

Is there a way to bypass this limit? why is it 4 MB when Cloudflare indicates a limit of 10 MB (https://developers.cloudflare.com/workers/platform/limits/)?

**Comment by james-elicx** [comment]
Unfortunately this error actually comes from the Vercel CLI and we are unable to do anything about it... 

https://github.com/vercel/vercel/blob/9290c57b83cc45a428e4ce96dd4402f97ec7f821/packages/next/src/edge-function-source/get-edge-function-source.ts#L86

You may need to patch the vercel dependency to disable it from throwing an error.

That does sound quite large though... Are you sure that such a big jump is right? It would have to be at least 0.45MB larger than it was before, as that's how far over the limit it is. I would not expect updating a dependency to push you that far over the limit if it was building fine before...

**Reply by SamyPesse** [reply]
Thanks @james-elicx ! We updated https://github.com/shikijs/shiki and it contains a lot of syntaxes, it might explain the important increase of size.

At first I thought code splitting solves this, but it looks like the Vercel CLI is looking at the size of the entire files together gzipped.
I'll look at maybe patching the Vercel CLI for now.

**Reply by simplenotezy** [reply]
@SamyPesse How did you resolve this issue? We are getting the same, after adding some very big JSON translation files to the project.

**Reply by simplenotezy** [reply]
@james-elicx some time have passed; did you manage to resolve it? Or do we need to manually patch the vercel dependency?

**Reply by SamyPesse** [reply]
@simplenotezy We patched vercel: https://github.com/GitbookIO/gitbook/blob/main/patches/%40vercel%252Fnext%404.3.6.patch
 with https://github.com/GitbookIO/gitbook/blob/main/package.json#L31-L33

We use `bun`, but I think it's possible to do the same with pnpm or npm.

This patch is also for a specific version of it; it has to be updated when we update next/vercel, but it's easy to do as it's a one-line patch.

**Reply by james-elicx** [reply]
@simplenotezy you would need to patch it, unfortunately we can't do anything about this in next-on-pages since it's part of Vercel's CLI.

**Reply by simplenotezy** [reply]
Thanks for prompt reply guys. I've managed to patch with [patch-package](https://www.npmjs.com/package/patch-package):

```diff
diff --git a/node_modules/@vercel/next/dist/index.js b/node_modules/@vercel/next/dist/index.js
index 5cb46cb..5241331 100644
--- a/node_modules/@vercel/next/dist/index.js
+++ b/node_modules/@vercel/next/dist/index.js
@@ -11811,7 +11811,7 @@ var import_path = require("path");
 // src/constants.ts
 var KIB = 1024;
 var MIB = 1024 * KIB;
-var EDGE_FUNCTION_SIZE_LIMIT = 4 * MIB;
+var EDGE_FUNCTION_SIZE_LIMIT = 10 * MIB;
 var DEFAULT_MAX_UNCOMPRESSED_LAMBDA_SIZE = 250 * MIB;
 var LAMBDA_RESERVED_UNCOMPRESSED_SIZE = 25 * MIB;
 ```

@james-elicx not sure, but are you also working on the OpenNext project? Wonder if it will be an issue here

**Reply by james-elicx** [reply]
> not sure, but are you also working on the OpenNext project? Wonder if it will be an issue here

Planning to. It won't be an issue there @simplenotezy - the OpenNext adaptor uses the Next.js build output and does not use the Vercel CLI.
